apis/addons/v1alpha1/addons_types.go

```diff
@@ -86,6 +87,35 @@ type MonitoringSpec struct {
 	// and it needs to be runing inside the namespace specified by `.monitoring.federation.namespace`
 	// with the service name 'prometheus'.
 	Federation *MonitoringFederationSpec `json:"federation,omitempty"`
+
+	// Settings For Monitoring Stack
+	// +optional
+	MonitoringStack *MonitoringStackSpec `json:"monitoringStack,omitempty"`
+}
+
+type MonitoringStackSpec struct {
+
+	// Enabled when set to true, deploys a MonitoringStack object
+	Enabled bool `json:"enabled,omitempty"`
+
+	// Settings for RHOBS Remote Write
+	// +optional
+	RHOBSRemoteWriteConfig *RHOBSRemoteWriteConfigSpec `json:"rhobsRemoteWriteConfig,omitempty"`
+
+	// Settings for the ServiceMonitor
+	// +optional
+	ServiceMonitorSpec *monv1.ServiceMonitorSpec `json:"serviceMonitorSpec,omitempty"`
```

Comment:
I think we should not embed the whole ServiceMonitorSpec here, because it is providing way too many options.
If we can't narrow it down, it might be better to have users still just dump them into their OLM bundles...